### ISSUE TITLE
User Login/Registration logic revamp

### ISSUE THREAD

**Issue by anubhav06** [anubhav06/eatit#14]
Currently the required fields for Registration are:
`Username, Password, Confirmation Password, Email, Mobile Number` and the user is logged in based on their `username` and `password`.

The username fields needs to be removed as it's kind of redundant keeping in mind what this project is i.e. a food ordering web application (with no use of a username).

User authentication should be done based on the `email` and `password` rather than `username` and `password`. 

This will be similar to restaurant's authentication, where restaurant owner's are authenticated based on their `email and password` and no username is asked during login/registration

**Comment by KendallDoesCoding** [comment]
> Currently the required fields for Registration are: `Username, Password, Confirmation Password, Email, Mobile Number` and the user is logged in based on their `username` and `password`.
> 
> The username fields needs to be removed as it's kind of redundant keeping in mind what this project is i.e. a food ordering web application (with no use of a username).
> 
> User authentication should be done based on the `email` and `password` rather than `username` and `password`.
> 
> This will be similar to restaurant's authentication, where restaurant owner's are authenticated based on their `email and password` and no username is asked during login/registration

This can be done with [Firebase](https://firebase.google.com/), I believe. Don't know if you've heard about it.

**Comment by anubhav06** [comment]
I don't think firebase is necessary for this. 
This can be done very easily using Django in the backend only.  
> This will be similar to restaurant's authentication, where restaurant owner's are authenticated based on their email and password and no username is asked during login/registration

**Comment by KendallDoesCoding** [comment]
> I don't think firebase is necessary for this. This can be done very easily using Django in the backend only.
> 
> > This will be similar to restaurant's authentication, where restaurant owner's are authenticated based on their email and password and no username is asked during login/registration

Ok cool. As I mentioned in #20 , I haven't really gotten into Django because my computer can't handle it... Just curious, does django give a password reset link or something to reset password, from the back-end, like Firebase does?

**Comment by anubhav06** [comment]
> > I don't think firebase is necessary for this. This can be done very easily using Django in the backend only.
> > > This will be similar to restaurant's authentication, where restaurant owner's are authenticated based on their email and password and no username is asked during login/registration
> 
> Ok cool. As I mentioned in #20 , I haven't really gotten into Django because my computer can't handle it... Just curious, does django give a password reset link or something to reset password, from the back-end, like Firebase does?

Yes, it does! That's one of the things which I love about Django that it makes user authentication very simple!
We can either reset the password as an admin from the Django Admin Panel, or let the client reset their password themselves (by providing them with a password reset link)

**Comment by KendallDoesCoding** [comment]
Oh. Very interesting
